R/read_csv.R

```diff
@@ -0,0 +1,239 @@
+    }
+    # read sampling data
+    if(sampling_info$sample_save_warmup == 1) {
+      num_of_draws <- sampling_info$sample_num_samples + sampling_info$sample_num_warmup
+    } else {
+      num_of_draws <- sampling_info$sample_num_samples
+    }
+    draws <- utils::read.csv(output_file, header = TRUE, comment.char = "#")
+    sampling_params_draws <- rbind(sampling_params_draws, draws[, sampling_info$sampler_params])
+    if(sampling_info$sample_save_warmup == 1) {
+      warmup_draws_array <- rbind(post_warmup_draws_array,
+                                  draws[1:sampling_info$sample_num_warmup, sampling_info$model_params])
+      post_warmup_draws_array <- rbind(post_warmup_draws_array,
+                                       draws[(sampling_info$sample_num_warmup+1):num_of_draws, sampling_info$model_params])
+    } else {
+      warmup_draws_array <- NULL
+      post_warmup_draws_array <- rbind(post_warmup_draws_array,
+                                       draws[, sampling_info$model_params])
+    }
+  }
+  #inverse mass matrix is returned separately
+  sampling_info$inverse_mass_matrix_diag <- NULL
+  if(!is.null(inverse_mass_matrix_diag)) {
+    dimnames(inverse_mass_matrix_diag) <- list(diagonal_elements = seq(dim(sampling_info$inverse_mass_matrix_diag)[1]),
+                                               chain_id = sampling_info$id)
+  }
+  sampling_info$model_params <- repair_variable_names(sampling_info$model_params)
+  num_chains <- length(sampling_info$id)
+  if(!is.null(warmup_draws_array)) {
+    warmup_draws_array <- posterior::as_draws_array(array(unlist(warmup_draws_array),
+                                                          dim = c(sampling_info$sample_num_warmup, num_chains, length(sampling_info$model_params)),
+                                                          dimnames = list(NULL, NULL, sampling_info$model_params)))
+  }
+  post_warmup_draws_array <- posterior::as_draws_array(array(unlist(post_warmup_draws_array),
+                                                             dim = c(sampling_info$sample_num_samples, num_chains, length(sampling_info$model_params)),
+                                                             dimnames = list(NULL, NULL, sampling_info$model_params)))
+  sampling_params_draws <- posterior::as_draws_array(array(unlist(sampling_params_draws),
+                                                             dim = c(sampling_info$sample_num_samples+sampling_info$sample_num_warmup, num_chains, length(sampling_info$sampler_params)),
+                                                             dimnames = list(NULL, NULL, sampling_info$sampler_params)))
+  list(
+    sampling_info = sampling_info,
+    inverse_mass_matrix_diag = inverse_mass_matrix_diag,
+    warmup = warmup_draws_array,
+    post_warmup = post_warmup_draws_array,
+    sampler = sampling_params_draws
+  )
+}
+
+# FIXME: also parse the csv header
+read_optim_csv <- function(output_file) {
```

Comment:
I moved both existent read_*_csv functions here. I suppose that is fine?

Reply:
Yeah, thanks!